File: SERVIDOR SOAP/ws_comer_dotnet_gr08/ec.edu.monster.service/ElectrodomesticoService.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using ec.edu.monster.db;
using ws_comer_dotnet_gr08;

namespace ec.edu.monster.service
{
    public class ElectrodomesticoService
    {
        public List<ElectrodomesticoResponse> ListarTodos()
        {
            var electrodomesticos = new List<ElectrodomesticoResponse>();

            using (var connection = DatabaseConnection.GetConnection())
            using (var cmd = connection.CreateCommand())
            {
                cmd.CommandText = "SELECT * FROM ELECTRODOMESTICO ORDER BY nombre";

                using (var reader = cmd.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        var e = new ElectrodomesticoResponse
                        {
                            IdElectrodomestico = Convert.ToInt32(reader["idElectrodomestico"]),
                            Nombre = Convert.ToString(reader["nombre"]),
                            PrecioVenta = Convert.ToDecimal(reader["precioVenta"])
                        };
                        electrodomesticos.Add(e);
                    }
                }
            }

            return electrodomesticos;
        }

        public ElectrodomesticoResponse BuscarPorId(int id)
        {
            using (var connection = DatabaseConnection.GetConnection())
            using (var cmd = connection.CreateCommand())
            {
                cmd.CommandText = "SELECT * FROM ELECTRODOMESTICO WHERE idElectrodomestico = @id";
                cmd.Parameters.AddWithValue("@id", id);

                using (var reader = cmd.ExecuteReader())
                {
                    if (reader.Read())
                    {
                        return new ElectrodomesticoResponse
                        {
                            IdElectrodomestico = Convert.ToInt32(reader["idElectrodomestico"]),
                            Nombre = Convert.ToString(reader["nombre"]),
                            PrecioVenta = Convert.ToDecimal(reader["precioVenta"])
                        };
                    }
                }
            }

            return null;
        }

        public ElectrodomesticoResponse Crear(ElectrodomesticoRequest request)
        {
            using (var connection = DatabaseConnection.GetConnection())
            using (var cmd = connection.CreateCommand())
            {
                cmd.CommandText = "INSERT INTO ELECTRODOMESTICO (nombre, precioVenta) VALUES (@nombre, @precioVenta)";
                cmd.Parameters.AddWithValue("@nombre", request.Nombre);
                cmd.Parameters.AddWithValue("@precioVenta", request.PrecioVenta);

                cmd.ExecuteNonQuery();

                cmd.CommandText = "SELECT CAST(SCOPE_IDENTITY() AS INT);";
                cmd.Parameters.Clear();

                var idObj = cmd.ExecuteScalar();
                if (idObj != null && idObj != DBNull.Value)
                {
                    var id = Convert.ToInt32(idObj);
                    return new ElectrodomesticoResponse
                    {
                        IdElectrodomestico = id,
                        Nombre = request.Nombre,
                        PrecioVenta = request.PrecioVenta
                    };
                }
            }

            return null;
        }

        public ElectrodomesticoResponse Actualizar(int id, ElectrodomesticoRequest request)
        {
            var existente = BuscarPorId(id);
            if (existente == null)
            {
                return null;
            }

            using (var connection = DatabaseConnection.GetConnection())
            using (var cmd = connection.CreateCommand())
            {
                cmd.CommandText = "UPDATE ELECTRODOMESTICO SET nombre = @nombre, precioVenta = @precioVenta WHERE idElectrodomestico = @id";
                cmd.Parameters.AddWithValue("@nombre", request.Nombre);
                cmd.Parameters.AddWithValue("@precioVenta", request.PrecioVenta);
                cmd.Parameters.AddWithValue("@id", id);

                var rows = cmd.ExecuteNonQuery();
                if (rows > 0)
                {
                    return new ElectrodomesticoResponse
                    {
                        IdElectrodomestico = id,
                        Nombre = request.Nombre,
                        PrecioVenta = request.PrecioVenta
                    };
                }
            }

            return null;
        }

        public bool Eliminar(int id)
        {
            var existente = BuscarPorId(id);
            if (existente == null)
            {
                return false;
            }

            using (var connection = DatabaseConnection.GetConnection())
            using (var checkCmd = connection.CreateCommand())
            {
                checkCmd.CommandText = "SELECT COUNT(*) FROM DETALLE_FACTURA WHERE idElectrodomestico = @id";
                checkCmd.Parameters.AddWithValue("@id", id);
                var count = Convert.ToInt32(checkCmd.ExecuteScalar());
                if (count > 0)
                {
                    return false;
                }
            }

            using (var connection = DatabaseConnection.GetConnection())
            using (var cmd = connection.CreateCommand())
            {
                cmd.CommandText = "DELETE FROM ELECTRODOMESTICO WHERE idElectrodomestico = @id";
                cmd.Parameters.AddWithValue("@id", id);
                var rows = cmd.ExecuteNonQuery();
                return rows > 0;
            }
        }
    }
}

File: 01.SERVIDOR/ws_comercializadora_gr08/src/main/java/ec/edu/monster/service/ElectrodomesticoService.java
package ec.edu.monster.service;

import ec.edu.monster.dto.ElectrodomesticoRequest;
import ec.edu.monster.dto.ElectrodomesticoResponse;
import ec.edu.monster.model.Electrodomestico;
import ec.edu.monster.db.DatabaseConnection;

import jakarta.enterprise.context.ApplicationScoped;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;

@ApplicationScoped
public class ElectrodomesticoService {
    
    private static final Logger LOGGER = Logger.getLogger(ElectrodomesticoService.class.getName());
    
    public List<ElectrodomesticoResponse> listarTodos() {
        List<ElectrodomesticoResponse> electrodomesticos = new ArrayList<>();
        
        try (Connection connection = DatabaseConnection.getConnection()) {
            String sql = "SELECT * FROM ELECTRODOMESTICO ORDER BY nombre";
            try (PreparedStatement stmt = connection.prepareStatement(sql)) {
                ResultSet rs = stmt.executeQuery();
                while (rs.next()) {
                    ElectrodomesticoResponse electrodomestico = new ElectrodomesticoResponse(
                        rs.getInt("idElectrodomestico"),
                        rs.getString("nombre"),
                        rs.getBigDecimal("precioVenta"),
                        rs.getString("fotoUrl")
                    );
                    electrodomesticos.add(electrodomestico);
                }
            }
        } catch (SQLException e) {
            LOGGER.log(Level.SEVERE, "Error listando electrodomésticos", e);
        }
        
        return electrodomesticos;
    }
    
    public ElectrodomesticoResponse buscarPorId(Integer id) {
        try (Connection connection = DatabaseConnection.getConnection()) {
            String sql = "SELECT * FROM ELECTRODOMESTICO WHERE idElectrodomestico = ?";
            try (PreparedStatement stmt = connection.prepareStatement(sql)) {
                stmt.setInt(1, id);
                ResultSet rs = stmt.executeQuery();
                if (rs.next()) {
                    return new ElectrodomesticoResponse(
                        rs.getInt("idElectrodomestico"),
                        rs.getString("nombre"),
                        rs.getBigDecimal("precioVenta"),
                        rs.getString("fotoUrl")
                    );
                }
            }
        } catch (SQLException e) {
            LOGGER.log(Level.SEVERE, "Error buscando electrodoméstico", e);
        }
        
        return null;
    }
    
    public ElectrodomesticoResponse crear(ElectrodomesticoRequest request) {
        try (Connection connection = DatabaseConnection.getConnection()) {
            String sql = "INSERT INTO ELECTRODOMESTICO (nombre, precioVenta, fotoUrl) VALUES (?, ?, ?)";
            try (PreparedStatement stmt = connection.prepareStatement(sql, PreparedStatement.RETURN_GENERATED_KEYS)) {
                stmt.setString(1, request.getNombre());
                stmt.setBigDecimal(2, request.getPrecioVenta());
                stmt.setString(3, request.getFotoUrl());
                
                int rowsAffected = stmt.executeUpdate();
                if (rowsAffected > 0) {
                    ResultSet rs = stmt.getGeneratedKeys();
                    if (rs.next()) {
                        Integer id = rs.getInt(1);
                        return new ElectrodomesticoResponse(id, request.getNombre(), request.getPrecioVenta(), request.getFotoUrl());
                    }
                }
            }
        } catch (SQLException e) {
            LOGGER.log(Level.SEVERE, "Error creando electrodoméstico", e);
        }
        
        return null;
    }
    
    public ElectrodomesticoResponse actualizar(Integer id, ElectrodomesticoRequest request) {
        try (Connection connection = DatabaseConnection.getConnection()) {
            // Verificar si el electrodoméstico existe
            ElectrodomesticoResponse existente = buscarPorId(id);
            if (existente == null) {
                return null;
            }
            
            String sql = "UPDATE ELECTRODOMESTICO SET nombre = ?, precioVenta = ?, fotoUrl = ? WHERE idElectrodomestico = ?";
            try (PreparedStatement stmt = connection.prepareStatement(sql)) {
                stmt.setString(1, request.getNombre());
                stmt.setBigDecimal(2, request.getPrecioVenta());
                stmt.setString(3, request.getFotoUrl());
                stmt.setInt(4, id);
                
                int rowsAffected = stmt.executeUpdate();
                if (rowsAffected > 0) {
                    return new ElectrodomesticoResponse(id, request.getNombre(), request.getPrecioVenta(), request.getFotoUrl());
                }
            }
        } catch (SQLException e) {
            LOGGER.log(Level.SEVERE, "Error actualizando electrodoméstico", e);
        }
        
        return null;
    }
    
    public boolean eliminar(Integer id) {
        try (Connection connection = DatabaseConnection.getConnection()) {
            // Verificar si el electrodoméstico existe
            ElectrodomesticoResponse existente = buscarPorId(id);
            if (existente == null) {
                return false;
            }
            
            // Verificar si tiene detalles de factura asociados
            String checkSql = "SELECT COUNT(*) FROM DETALLE_FACTURA WHERE idElectrodomestico = ?";
            try (PreparedStatement checkStmt = connection.prepareStatement(checkSql)) {
                checkStmt.setInt(1, id);
                ResultSet rs = checkStmt.executeQuery();
                if (rs.next() && rs.getInt(1) > 0) {
                    LOGGER.log(Level.WARNING, "No se puede eliminar electrodoméstico con detalles de factura asociados");
                    return false;
                }
            }
            
            String sql = "DELETE FROM ELECTRODOMESTICO WHERE idElectrodomestico = ?";
            try (PreparedStatement stmt = connection.prepareStatement(sql)) {
                stmt.setInt(1, id);
                int rowsAffected = stmt.executeUpdate();
                return rowsAffected > 0;
            }
        } catch (SQLException e) {
            LOGGER.log(Level.SEVERE, "Error eliminando electrodoméstico", e);
        }
        
        return false;
    }
}
